validator/src/main.rs

```diff
@@ -2443,7 +2440,6 @@ pub fn main() {
         accounts_db_test_hash_calculation: matches.is_present("accounts_db_test_hash_calculation"),
         accounts_db_config,
         accounts_db_skip_shrink: matches.is_present("accounts_db_skip_shrink"),
-        accounts_db_use_index_hash_calculation: matches.is_present("accounts_db_index_hashing"),
```

Comment:
let's query this outside the config initialization and log a deprecation warning and removal suggestion, in lieu of the updated help strings.

https://github.com/solana-labs/solana/pull/23181/files#diff-59e2068dd4d49c21ab87f3a5d8792d2daca826ab6924b32413e551c0e3035f61 for inspiration